zkevm-circuits/src/evm_circuit/util/common_gadget/tx_eip2930.rs

```diff
@@ -0,0 +1,134 @@
+                let [address_len, storage_key_len, gas_cost] = [
+                    TxFieldTag::AccessListAddressesLen,
+                    TxFieldTag::AccessListStorageKeysLen,
+                    TxFieldTag::AccessListGasCost,
+                ]
+                .map(|field_tag| cb.tx_context(tx_id.expr(), field_tag, None));
+
+                // Let copy-circuit to write the tx-table's access list addresses into rw-table.
+                cb.copy_table_lookup(
+                    tx_id.expr(),
+                    CopyDataType::AccessListAddresses.expr(),
+                    tx_id.expr(),
+                    CopyDataType::AccessListAddresses.expr(),
+                    0.expr(),
+                    address_len.expr(),
+                    0.expr(),
+                    address_len.expr(),
+                    0.expr(),
+                    // TODO
+                    0.expr(),
+                );
+
+                // Let copy-circuit to write the tx-table's access list storage keys into rw-table.
+                cb.copy_table_lookup(
+                    tx_id.expr(),
+                    CopyDataType::AccessListStorageKeys.expr(),
+                    tx_id.expr(),
+                    CopyDataType::AccessListStorageKeys.expr(),
+                    0.expr(),
+                    storage_key_len.expr(),
+                    0.expr(),
+                    storage_key_len.expr(),
+                    0.expr(),
+                    // TODO
+                    0.expr(),
+                );
+
+                [address_len, storage_key_len, gas_cost]
+            });
+
+        Self {
+            is_eip2930_tx,
+            access_list_address_len,
+            access_list_storage_key_len,
+            access_list_gas_cost,
+        }
+    }
+
+    pub(crate) fn assign(
+        &self,
+        region: &mut CachedRegion<'_, '_, F>,
+        offset: usize,
+        tx: &Transaction,
+    ) -> Result<(), Error> {
+        self.is_eip2930_tx.assign(
+            region,
+            offset,
+            F::from(tx.tx_type as u64),
+            F::from(TxType::Eip2930 as u64),
+        )?;
+
+        let (access_list_address_len, access_list_storage_key_len) =
+            access_list_size(&tx.access_list);
+
+        self.access_list_address_len.assign(
+            region,
+            offset,
+            Value::known(F::from(access_list_address_len)),
+        )?;
+        self.access_list_storage_key_len.assign(
+            region,
+            offset,
+            Value::known(F::from(access_list_storage_key_len)),
+        )?;
+        self.access_list_gas_cost.assign(
+            region,
+            offset,
+            Value::known(F::from(tx.access_list_gas_cost)),
+        )?;
+
+        Ok(())
+    }
+
+    pub(crate) fn gas_cost(&self) -> Expression<F> {
+        select::expr(
+            self.is_eip2930_tx.expr(),
+            self.access_list_gas_cost.expr(),
+            0.expr(),
+        )
```

Comment:
```suggestion
        select::expr(
            self.is_eip2930_tx.expr(),
            self.access_list_address_len.expr() * 1900.expr()
                + self.access_list_storage_key_len.expr() * 2400.expr(),
            0.expr(),
        )
```

However, please use the `1900` and `2400` from `eth_types::evm_types::GasCost`

Reply:
Yes, fixed it in commit [6880378](https://github.com/scroll-tech/zkevm-circuits/pull/1045/commits/68803783b1447b8fefe2c1fabd626beae51b8a83):
```
access_list_address_len.expr() * GasCost::ACCESS_LIST_PER_ADDRESS.expr()
    + access_list_storage_key_len.expr() * GasCost::ACCESS_LIST_PER_STORAGE_KEY.expr()
```